lib/gui/app/components/source-selector/source-selector.tsx

```diff
@@ -73,36 +77,158 @@ function getState() {
 	};
 }
 
-interface ImageSelectorProps {
+const URLSelector = ({ done }: { done: (imageURL: string) => void }) => {
+	const [imageURL, setImageURL] = React.useState('');
+	const [recentImages, setRecentImages]: [
+		string[],
+		(value: React.SetStateAction<string[]>) => void,
+	] = React.useState([]);
+	React.useEffect(() => {
+		const fetchRecentUrlImages = async () => {
+			try {
+				const recentUrlImages: any = await jsonStorage.get('recentUrlImages');
+				setRecentImages(recentUrlImages);
+			} catch (err) {
+				console.error(err);
+			}
+		};
+		fetchRecentUrlImages();
+	}, []);
+	return (
+		<Modal
+			done={async () => {
+				const sanitizedRecentUrls = _.uniq(
+					_.reject([...recentImages, imageURL], _.isEmpty),
```

Comment:
This will not work if `recentImages` is `undefined`.

Reply:
It won't be, the default returned value is an empty object

Reply:
In case it fails, it still is `[]` with React's `useState`

Reply:
See the screenshot above